autoload/go/issue.vim

```diff
@@ -0,0 +1,34 @@
+let s:templatepath = go#util#Join(fnamemodify('<sfile>', ':p:h'), '.github', 'ISSUE_TEMPLATE.md')
+
+function! go#issue#New() abort
+  let body = substitute(s:issuebody(), '[^A-Za-z0-9_.~-]', '\="%".printf("%02X",char2nr(submatch(0)))', 'g')
+  let url = "https://github.com/fatih/vim-go/issues/new?body=" . l:body
+  call go#tool#OpenBrowser(l:url)
+endfunction
+
+function! s:issuebody() abort
+  let lines = readfile(s:templatepath)
+
+  let rtrimpat = '[[:space:]]\+$'
+  let body = []
+  for l in lines
+    let body = add(body, l)
+
+    if l =~ '^\* Vim version'
+      redir => out
+        silent version
+      redir END
```

Comment:
It is probably better to use `let l:out = execute('silent version')` instead of `redir`. `redir` looks nice, but there some issues when you run multiple `redir`s at the same time (e.g. nesting, from autocmds) which can result in very strange and confusing errors.  
Since we don't control the entire setup (other plugins, people's vimrcs, etc.) it's usually best to avoid `redir` unless there is no other way.

Reply:
Under what conditions would there be another `redir` being executed?

Reply:
Given that vim 7.4 doesn't have `execute()`, that doesn't seem like a viable option until we drop support for vim 7.4.

Reply:
Oh, that was added in 7.4.2008, which is just a bit after the 7.4.1689 we support – thought it was earlier.

> Under what conditions would there be another redir being executed?

Mostly in cases with autocmds or another function calling this. You can never really know, that's the thing.